webapi/src/main/java/org/sakaiproject/webapi/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,96 @@
+import java.util.Map;
+
+@Slf4j
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(MissingSessionException.class)
+    public ResponseEntity<Object> handleMissingSessionException(MissingSessionException ex, WebRequest request) {
+        return createErrorResponse(HttpStatus.FORBIDDEN, "Missing or invalid Sakai session", ex);
+    }
+
+    @ExceptionHandler(UnknownSiteException.class)
+    public ResponseEntity<Object> handleUnknownSiteException(UnknownSiteException ex, WebRequest request) {
+        return createErrorResponse(HttpStatus.BAD_REQUEST, "Unknown Sakai site", ex);
+    }
+
+    @ExceptionHandler(ForbiddenAccessException.class)
+    public ResponseEntity<Object> handleForbiddenAccessException(ForbiddenAccessException ex, WebRequest request) {
+        return createErrorResponse(HttpStatus.FORBIDDEN, "Access forbidden", ex);
+    }
+
+    @ExceptionHandler(PermissionException.class)
+    public ResponseEntity<Object> handlePermissionException(PermissionException ex, WebRequest request) {
+        return createErrorResponse(HttpStatus.FORBIDDEN, "Permission denied", ex);
+    }
+
+    @ExceptionHandler(IdUnusedException.class)
+    public ResponseEntity<Object> handleIdUnusedException(IdUnusedException ex, WebRequest request) {
+        return createErrorResponse(HttpStatus.NOT_FOUND, "Resource not found", ex);
+    }
+
+    @ExceptionHandler(InUseException.class)
+    public ResponseEntity<Object> handleInUseException(InUseException ex, WebRequest request) {
+        return createErrorResponse(HttpStatus.CONFLICT, "Resource is currently in use", ex);
+    }
+
+    @ExceptionHandler(TypeException.class)
+    public ResponseEntity<Object> handleTypeException(TypeException ex, WebRequest request) {
+        return createErrorResponse(HttpStatus.BAD_REQUEST, "Invalid resource type", ex);
+    }
+
+    @ExceptionHandler(ServerOverloadException.class)
+    public ResponseEntity<Object> handleServerOverloadException(ServerOverloadException ex, WebRequest request) {
+        return createErrorResponse(HttpStatus.SERVICE_UNAVAILABLE, "Server is currently overloaded", ex);
+    }
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<Object> handleAllUncaughtException(Exception ex, WebRequest request) {
+        log.error("Uncaught exception", ex);
+        return createErrorResponse(HttpStatus.INTERNAL_SERVER_ERROR, "An unexpected error occurred", ex);
+    }
+
+    private ResponseEntity<Object> createErrorResponse(HttpStatus status, String message, Exception ex) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", status.value());
+        body.put("error", status.getReasonPhrase());
+        body.put("message", message);
+        
+        if (ex.getMessage() != null) {
+            body.put("detail", ex.getMessage());
+        }
+        
+        log.error("{}: {}", message, ex.getMessage(), ex);
```

Comment:
```suggestion
        log.error("{}: {}", message, ex.toString());
```

Reply:
I don't think you need getMessage and the exception, toString should cover the two.

Reply:
ha agree.... we were just editing this on the core call